### PR TITLE
ReporterCommand: Print the number of successfully created reports

### PR DESCRIPTION
**Pull request by sschuberth**
To quickly give an overview of how bad a failed reporter run was.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

### CODE REVIEW

File: cli/src/main/kotlin/commands/ReporterCommand.kt [hunk 1]
```diff
@@ -232,17 +232,18 @@ class ReporterCommand : CliktCommand(
             }
         }
 
-        reportDurationMap.forEach { (entry, duration) ->
+        val successCount = reportDurationMap.count { (entry, duration) ->
             val (reporter, file) = entry
-            if (file.isFile) {
-                println(
+            file.isFile.also {
+                if (it) println(
```
Comment (mnonnenmacher) [inline cli/src/main/kotlin/commands/ReporterCommand.kt:238]: If the if spans multiple lines it should always use curly braces.

File: cli/src/main/kotlin/commands/ReporterCommand.kt [hunk 2]
```diff
@@ -232,17 +232,18 @@ class ReporterCommand : CliktCommand(
             }
         }
 
-        reportDurationMap.forEach { (entry, duration) ->
+        val successCount = reportDurationMap.count { (entry, duration) ->
             val (reporter, file) = entry
-            if (file.isFile) {
-                println(
+            file.isFile.also {
```
Comment (mnonnenmacher) [inline cli/src/main/kotlin/commands/ReporterCommand.kt:237]: Why do we actually need this check? Is it possible that a reporter does not throw an exception and not create the file at the same time?

File: cli/src/main/kotlin/commands/ReporterCommand.kt (hunk 1 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/ReporterCommand.kt:238]: Hmm, but it's not the `if` that spans multiple lines, it's just the `println` argument (like already before).

File: cli/src/main/kotlin/commands/ReporterCommand.kt (hunk 2 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/ReporterCommand.kt:237]: > Is it possible that a reporter does not throw an exception and not create the file at the same time?

Probably not, I'll look into changing this with a rework of this commit.

Comment (sschuberth) [reply cli/src/main/kotlin/commands/ReporterCommand.kt:237]: Now I recall the reason: The "Successfully created [...]" status lines should be printed en-bloc after all reports were generated, for a better overview. And at that time the easiest way to check whether the creation of a report was successful is to check for its output file.

File: cli/src/main/kotlin/commands/ReporterCommand.kt (hunk 1 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/ReporterCommand.kt:238]: (Actually, I admit I'm using this edge-case to "cheat" a bit as I don't want to wrap the string argument to `println`, which I would need to do if I added further indentation 😉)

File: cli/src/main/kotlin/commands/ReporterCommand.kt (hunk 2 above)
Comment (mnonnenmacher) [reply cli/src/main/kotlin/commands/ReporterCommand.kt:237]: Ok, but I thought if the reporter failed there would be no entry in `reportDurationMap`.

File: cli/src/main/kotlin/commands/ReporterCommand.kt (hunk 1 above)
Comment (mnonnenmacher) [reply cli/src/main/kotlin/commands/ReporterCommand.kt:238]: The Kotlin coding conventions have no opinion on that, so I'll accept the cheating ;)

File: cli/src/main/kotlin/commands/ReporterCommand.kt (hunk 2 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/ReporterCommand.kt:237]: Doh, you're right. Let me fix that.

File: cli/src/main/kotlin/commands/ReporterCommand.kt [hunk 3]
```diff
@@ -237,8 +237,9 @@ class ReporterCommand : CliktCommand(
             println("Successfully created the '${reporter.reporterName}' report at '$file' in ${duration.inSeconds}s.")
         }
 
+        println("Successfully created ${reportDurationMap.size} of ${reports.size} report(s).")
```
Comment (mnonnenmacher) [inline cli/src/main/kotlin/commands/ReporterCommand.kt:240]: I think there should be a different message if no report was created successfully.

Comment (sschuberth) [reply cli/src/main/kotlin/commands/ReporterCommand.kt]: Done...

**Pull request merged**